framework-features-prio (#11237)
getsentry/sentry-docs: docs/platforms/javascript/guides/nuxt/features/component-tracking.mdx

```javascript
Sentry.init({
  // ...
  trackComponents: true,
  // OR
  trackComponents: [
    "App",
    "RwvHeader",
    "RwvFooter",
    "RwvArticleList",
    "Pagination",
  ],
});
```

The default is `false`.

#### `hooks`

Control which lifecycle hooks should be tracked. This is helpful if, for example, you want to know if some components are removed during the initial page load, in which case you can add a `unmount` hook to the default:

```javascript
Sentry.init({
  // ...
  trackComponents: true
  hooks: ["mount", "update", "unmount"],
});
```

The following hooks are available to track in Vue 3: `['activate', 'create', 'unmount', 'mount', 'update']`

Note that when specifying `hooks`, we use the simple verb rather than `before` and `-ed` pairs. For example, `unmount` is correct, while `beforeUnmount` and `unmounted` are incorrect.

<Alert>

In Vue 2, use `destroy` instead of `unmount`. `destroy` does not work in Vue 3, as the names of the lifecycle hooks themselves [changed](https://v3-migration.vuejs.org/breaking-changes/#other-minor-changes) in Vue 3.

</Alert>

The default set of hooks is `['activate', 'mount', 'update']`.

#### `timeout`

> You can specify how long the root rendering span should wait for the last component to render.
> Every new rendering cycle debounces the timeout, and it starts counting from the beginning. Once the timeout is reached, tracking is completed, and all the rendering information is sent to Sentry:

Reviewer: ```suggestion
You can specify how long the root rendering span should wait until the last component is rendered.
Every new rendering cycle debounces the timeout and it starts counting from the beginning. Once the timeout is reached, tracking is completed and all the rendering information is sent to Sentry:
```

```javascript
Sentry.init({
  // ...
  trackComponents: true,
  timeout: 500,
});
```

The default is `2000`.

#### Alternative Configuration With `tracingOptions`

> You can also group the component tracking options by using the optional `tracingOptions` property in `Sentry.init`:

Reviewer: ```suggestion
You can also group the component-tracking options by using the optional `tracingOptions` property in `Sentry.init`:
```

```javascript
Sentry.init({
  // ...
  tracingOptions: {
    trackComponents: true;
    timeout: 500;
    hooks: ['mount', 'update'];
  }
})
```

> Note that when you use this property there is no change in behaviour, as opposed to when you use the three top-level properties described above.

Reviewer: ```suggestion
Note, that when you use this property there is no change in behaviour, as opposed to when you use the three top-level properties described above.
```
